### PR TITLE
#2453: Enable AA JWT auth

### PR DESCRIPTION
**Pull request by johnnymetz**
## What does this PR do?

- Part of #4582 

## Reviewer Tips

- Corresponding App PR: https://github.com/pixiebrix/pixiebrix-app/pull/2501/files

## Checklist

- [x] Run Storybook and manually confirm that all stories are working
- [x] Designate a primary reviewer: @twschiller

### CODE REVIEW

File: src/options/Navbar.tsx [hunk 1]
```diff
@@ -34,7 +34,19 @@ import { ThemeLogo } from "@/utils/themeUtils";
 
 const Navbar: React.FunctionComponent<{ logo: ThemeLogo }> = ({ logo }) => {
   const { email } = useSelector(selectAuth);
-  const [serviceURL] = useAsyncState<string>(getBaseURL);
+  const [serviceURL] = useAsyncState<string>(async () => {
+    const baseURL = (await getBaseURL()) ?? DEFAULT_SERVICE_URL;
+    const partnerAuth = await readPartnerAuthData();
+    let url;
```
Comment (twschiller) [inline src/options/Navbar.tsx:40]: NIT: Prefer ternary operator to let

File: src/options/Navbar.tsx [hunk 2]
```diff
@@ -81,11 +93,7 @@ const Navbar: React.FunctionComponent<{ logo: ThemeLogo }> = ({ logo }) => {
 
         <ul className="navbar-nav navbar-nav-right flex-grow-1 justify-content-end">
           {serviceURL && (
-            <Nav.Link
-              className="px-3"
-              target="_blank"
-              href={serviceURL ?? DEFAULT_SERVICE_URL}
-            >
+            <Nav.Link className="px-3" target="_blank" href={serviceURL}>
```
Comment (twschiller) [inline src/options/Navbar.tsx:96]: The href will be null while the async state is resolving. I think you still want to have the `?? DEFAULT_SERVICE_URL` here

File: src/options/Navbar.tsx [hunk 3]
```diff
@@ -34,7 +34,19 @@ import { ThemeLogo } from "@/utils/themeUtils";
 
 const Navbar: React.FunctionComponent<{ logo: ThemeLogo }> = ({ logo }) => {
   const { email } = useSelector(selectAuth);
-  const [serviceURL] = useAsyncState<string>(getBaseURL);
+  const [serviceURL] = useAsyncState<string>(async () => {
```
Comment (twschiller) [inline src/options/Navbar.tsx:37]: I think we need to create a follow up ticket to listen for changes to the partner auth data. I suspect on initial login that this async state won't refresh?

File: src/options/Navbar.tsx [hunk 4]
```diff
@@ -34,7 +34,19 @@ import { ThemeLogo } from "@/utils/themeUtils";
...
+    const baseURL = (await getBaseURL()) ?? DEFAULT_SERVICE_URL;
+    const partnerAuth = await readPartnerAuthData();
+    let url;
+
+    if (partnerAuth?.token) {
```
Comment (twschiller) [inline src/options/Navbar.tsx:42]: 👍

**Review by twschiller — APPROVED** [review]
See comment on potentially needing a followup ticket to listen for state changes

File: src/options/Navbar.tsx (hunk 3 above)
Comment (johnnymetz) [reply src/options/Navbar.tsx:37]: Ticket created and added as comment

**Pull request merged**